### DISCUSSION TITLE
how Multi threaded or coroutine isolation？？

### DISCUSSION THREAD

**Discussion by tianshangwuyun** [microsoft/autogen#4312] [Q&A]
How to perform thread isolation in version 0.4? Usually, we build a multi-agent application that has many users accessing it. How to isolate requests from different users? Langgraph uses the threaded mechanism, and what method autogen uses. We hope for guidance. No relevant introduction was found in the document

**Comment by ekzhu** [comment]
Multi-tenant isolation can be done through topic for the Core API. The SingleThreadedAgentRuntime is coroutine-based and separate different user sessions through instantiation of different agent instances, by publishing to different topic IDs. 

You can take a look at the Core documentation: Core Concepts -> topics and subscriptions. 

For AgentChat each user session needs to instantiate new agent and teams. But we are working to address this so you can use existing agents and teams already registered with the runtime.
